### ISSUE TITLE
d2i_PrivateKey() fails on EVP_PKEY_EC after upgrading OpenSSL 1.1.1a to 1.1.1k

### ISSUE THREAD

**Issue by Iniesta8** [openssl/openssl#16380]
After upgrading OpenSSL from 1.1.1a to 1.1.1k, I noticed failed unit tests in connection with `d2i_PrivateKey()` calls. I want to read DER encoded key data from a buffer into a `EVP_PKEY` structure.

The problematic code line:

```
EVP_PKEY* key = d2i_PrivateKey(EVP_PKEY_EC, NULL, (const unsigned char**)&key_data, key_data_len);
```

The call is failing (returning `NULL`) and the following errors are set:

```
error:0D0680A8:asn1 encoding routines:asn1_check_tlen:wrong tag
error:0D06C03A:asn1 encoding routines:asn1_d2i_ex_primitive:nested asn1 error
```

I noticed a change in documentation regarding `d2i_PrivateKey()`. The following paragraph was added:

```
To decode a key with type EVP_PKEY_EC, d2i_PublicKey() requires *a to be a non-NULL EVP_PKEY structure assigned an EC_KEY structure referencing the proper EC_GROUP.
```

I'm afraid that's my problem...right?

And if so, how to call the function correctly when the underlying curve is ED25519?

**Comment by Iniesta8** [comment]
Okay, it works using `d2i_AutoPrivateKey()`...

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input.
